### ISSUE TITLE
why not use the import

### ISSUE THREAD

**Issue by matrix3d** [Corsaair/redtamarin#10]
i have test the redtamarin sometime,and it not easy use.it is use include ,not use the import.
why not use the java like public static void main?and full support the import

**Comment by zwetan** [comment]
you have to use both the include and import, and import are fully supported AFAIK.

see [Compile with the Flex SDK](https://github.com/Corsaair/redtamarin/wiki/Compile-with-the-Flex-SDK)
where an alternative way is explained

about 

> why not use the java like public static void main?

because we are not using Java we are using ActionScript 3.0,
not the same language and not the same rules.

also, it's IMHO better to not have to depend on a main function
for when you want to use AS3 in shell scripts for example.

**Comment by matrix3d** [comment]
can you see my code?[Compiler] Error #1180: Call to a possibly undefined method Test.

i just use the redtamarin-sdk-windows\bin redbean.exe

the build.as3

<pre lang='as'>import redbean.*;
compile("Main.as");
//shell("Main.as");
projector("Main",false,null,["Main.abc"]);</pre>

the Main.as

<pre lang='as'>package
{
    public class Main
    {
        public function Main()
        {
            new Test();
        }
    }
    new Main();
}</pre>

the Test.as

<pre lang='as'>package
{
    public class Test
    {
        public function Test()
        {
            trace( "hello world test" );
        }
    }
}</pre>

**Comment by zwetan** [comment]
that's normal, you did not use the **include** directive
so the file is not found by the compiler

you HAVE TO use includes, see
[AS3 statement: include](http://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/statements.html#include)

file: **includes.as**

``` as3
include "Test.as";
//other file to include 
//other file to include
//etc.
```

file: **Main.as**

``` as3
include "includes.as";

package
{
    public class Main
    {
        public function Main()
        {
            new Test();
        }
    }
    new Main();
}
```

or alternatively

file: **myprogram.as**

``` as3
include "includes.as";

new Main();
```

and file: **build.as3**

``` as3
import redbean.*;
compile("myprogram.as");
//shell("myprogram.as");
projector("myprogram",false,null,["myprogram.abc"]);
```

**Comment by matrix3d** [comment]
hello.this is my say,the lib not full support the import.
it is need include.if the norm swf.this is not need the include.so is it need auto include in the source dir,or same package dir?a big as3 lib,incliude many the as?

**Comment by matrix3d** [comment]
I have run succees with your code,not bad.but the include not easy vs the import.

and compile("myprogram.as") run good.

but why shell("myprogram.as") with error?

**Comment by zwetan** [comment]
> this is my say,the lib not full support the import.

include and import are 2 different things

**import** is part of the AS3 language, it will always work
take for example something that is already embedded in the runtime

``` as3
import C.unistd.*;

trace( getlogin() );
```

it works, even if you don't have to include `/src/C/unistd.as`

**include** is more related to how the compiler compile the code,
MXMLC will go trough the main class and deduce which files to include based on the imports,
but ASC does not do that, that's why you have to manage the include yourself

in `redbean` you have an utility function `generateIncludes()` that can help you generate those includes if you don't want to do it by hand, look at the doc there is an example
http://docs.redtamarin.com/redbean/latest/redbean/package.html#generateIncludes()

> why shell("myprogram.as") with error?

you need to run the compiled file
eg. `shell("myprogram.abc")`

---

Now, I'm gonna be blunt and please don't take that the wrong way, but it is your job as a developer to understand how to compile the code.

Yes it is not easy at first but that's what developers do.

So I'm not saying "you're not a developer" here, but when I read

> your lib does not fully support imports

I understand that instead

> as a developer I never bothered to understand how the compiler works
> so now when it does not do what I think it should do,
> it's easier for me to blame the tool than to learn
> how the compiler actually works

between redbean and the doc about using the Flex SDK to compile code
there is not much more I can do

in other words, you can complain all you want, the problem is still about you
learning how the compiler works.

**Comment by matrix3d** [comment]
yes,you are right

**Comment by zwetan** [comment]
I will try to add more documentation about that, cheers
